### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by fwpushan**
1. Add all unit test cases for user route
2. Fix unit test defect for global error handling middleware of express.

### CODE REVIEW

File: __mocks__/passport.js [hunk 1]
```diff
@@ -33,13 +33,19 @@ const user = {
...
+};
+passport.clearGlobal = (key) => { delete passport.global[key]; };
 
 function authenticate(strategy, options) {
   'use strict';
 
   return (req, res, next) => {
     req.user = passport.aUser;
     req.isAuthenticated = () => true; // Skip calling jwtStrategy, auth the request straight
+    // req.body = passport.global[req.originalUrl];
```
Comment (micheal-w-wells) [inline __mocks__/passport.js]: probably shouldn't include this commented out code in commit

Comment (fwpushan) [reply __mocks__/passport.js]: sure missed that

**Comment by micheal-w-wells** [comment]
My bad thought i was only reviewer on this one @fwpushan @jleach .  Poorly executed multi-tasking.  @Kyubinhan has since told me about the rule where PR owner merges after reviews as well so I'll be sure to follow that.

**Comment by jleach** [comment]
Ya, Best if the owner merges so they can deal with any fallout from merge conflicts. Its a bit different if you Fork a repo, in which case you usually don’t have access to merge so the maintainer will do it.

J.

**Pull request merged**